libnavigation-core/src/main/java/com/mapbox/navigation/core/predownloading/TimeBasedNextVoiceInstructionsProvider.kt

```diff
@@ -0,0 +1,74 @@
+package com.mapbox.navigation.core.predownloading
+
+import com.mapbox.api.directions.v5.models.VoiceInstructions
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.utils.internal.logW
+
+internal class TimeBasedNextVoiceInstructionsProvider(
+    private val observableTimeSeconds: Int,
+) : NextVoiceInstructionsProvider {
+
+    override fun getNextVoiceInstructions(
```

Comment:
I was thinking about proxying this work to a background thread but I measured the time required for these calculations and results are the following: for the route from Munich to Toledo with observableTime being equal to 1 hour the maximum time during which this method was executing is 3 ms. Considering that the route is quite long and the default observableTim will be much less (3 minutes), I think we're good to leave these calculations on the main thread.